### PR TITLE
feat(get-jwks): add code to thrown error (#156)

### LINKED ISSUE #156: Feature request: More specific errors
Currently, `get-jwks` throws generic instances of `Error` and all details are "hidden" in the message (and are thus not easily accessible). This is problematic because it lumps together all errors that can potentially occur:

- upstream errors (e.g., the discovery endpoint returns a 503)
- errors caused by the library itself (not really an issue here as far as I can tell)
- downstream errors (errors for which arguably the client is responsible, e.g. `JWK_NOT_FOUND`)

This matters because the HTTP status code returned to the client should indicate the root cause and be 502, 500 and 400 (or 401), respectively. Mapping this correctly is currently not feasible.

I propose to  use the [`code`](https://nodejs.org/api/errors.html#errorcode) property to convey that information. We could, at least in part, use the keys in the [message mapping](https://github.com/nearform/get-jwks/blob/main/src/get-jwks.js#L7) for that. Defining a new error class, say `JwksError`, might also be beneficial to simplify user land code.

I'm more than happy to whip up a PR if you are in principle willing to add such a feature.

### PR DESCRIPTION
**Pull request by pigulla**
It's been quite a while since I wrote vanilla JavaScript, so please feel free to suggest improvements.

I'm not really happy with the tests, I didn't see a way with tap to assert the class of an error and only do a partial match of its properties. Suggestions are welcome!

Closes #156

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "get-jwks",
-  "version": "7.1.1",
+  "version": "8.0.0",
```
Comment (simoneb) [inline package.json:3]: don't bump the version, we do that when releasing a new version

**Review by simoneb — COMMENTED** [review]
The approach looks good, but have you considered using fastify-error as suggested by @mcollina in his comment to the issue?

**Comment by pigulla** [comment]
> The approach looks good, but have you considered using fastify-error as suggested by @mcollina in his comment to the issue?

Yes, I looked into that, but I didn't see a good way to add additional properties to the error (specifically `response` and `body` for failed HTTP requests). They can be added manually, of course, but that feels rather clunky.

In any case, if that is the preferred way of doing this, I'm happy to update the PR accordingly.

File: package.json [hunk 2]
```diff
@@ -35,8 +35,8 @@
   },
   "devDependencies": {
     "@fastify/jwt": "^6.1.0",
-    "@types/lru-cache": "^7.4.0",
```
Comment (simoneb) [inline package.json:38]: why did you remove this?

File: test/getJwkDiscovery.spec.js [hunk 3]
```diff
@@ -36,7 +39,11 @@ t.test('rejects if the request fails', async t => {
   const [{ alg, kid }] = jwks.keys
   const getJwks = buildGetJwks({ providerDiscovery: true })
 
-  const expectedError = new Error('Internal Server Error')
+  const expectedError = {
```
Comment (simoneb) [inline test/getJwkDiscovery.spec.js:42]: why do you need to create the error in this way rather than creating an instance of the error?

File: package.json (hunk 2 above)
Comment (pigulla) [reply package.json:38]: It is no longer needed:
> npm WARN deprecated @types/lru-cache@7.10.10: This is a stub types definition. lru-cache provides its own type definitions, so you do not need this installed.

File: test/getJwkDiscovery.spec.js (hunk 3 above)
Comment (pigulla) [reply test/getJwkDiscovery.spec.js:42]: That's the issue I hinted at. I did not see a way to create an instance of that error that passes the built-in matcher because that would require me to provide the correct `response` object:
```typescript
const expectedError = new GetJwksError(errorCode.JWKS_REQUEST_FAILED, {
  response: new Response({}, { timeout: 5000 }),
  body: { msg: 'boom' },
})
```
...and that just didn't feel right because it is then asserting some internals that should not be relevant.

In other test frameworks I would do something like this:
```typescript
const expectedError = new GetJwksError(errorCode.JWKS_REQUEST_FAILED, {
  response: expect.anyt(Response),
  body: { msg: 'boom' },
})
```
but I did not see a way to do this with `tap`.

**Review by simoneb — APPROVED** [review]
